gtest_parallel.py

```diff
@@ -358,6 +364,37 @@ def dump_to_file_and_close(self):
 
 # Record of test runtimes. Has built-in locking.
 class TestTimes(object):
+  class LockedFile(object):
+    def __init__(self, filename, mode):
+      self._filename = filename
+      self._mode = mode
+      self._fo = None
+
+    def __enter__(self):
```

Comment:
Can we put some kind of assert here that we're calling this from the main thread? I think we only read/write test times from the main thread, right?